Jenkinsfile

```diff
@@ -0,0 +1,196 @@
+                        }
+                    }
+                    post {
+                        always {
+                            bbcGithubNotify(context: "tests/py27", status: env.py27_result)
+                        }
+                    }
+                }
+                stage ("Python 3 Unit Tests") {
+                    steps {
+                        script {
+                            env.py3_result = "FAILURE"
+                        }
+                        bbcGithubNotify(context: "tests/py3", status: "PENDING")
+                        // Use a workdirectory in /tmp to avoid shebang length limitation
+			withBBCRDPythonArtifactory {
+                            sh 'tox -e py3 --recreate --workdir /tmp/$(basename ${WORKSPACE})/tox-py3'
+			}
+                        script {
+                            env.py3_result = "SUCCESS" // This will only run if the sh above succeeded
+                        }
+                    }
+                    post {
+                        always {
+                            bbcGithubNotify(context: "tests/py3", status: env.py3_result)
+                        }
+                    }
+                }
+                stage ("Debian Source Build") {
+                    steps {
+                        script {
+                            env.debSourceBuild_result = "FAILURE"
+                        }
+                        bbcGithubNotify(context: "deb/sourceBuild", status: "PENDING")
+
+                        sh 'rm -rf deb_dist'
+                        sh 'python ./setup.py sdist'
+                        sh 'make dsc'
+                        bbcPrepareDsc()
+                        stash(name: "deb_dist", includes: "deb_dist/*")
+                        script {
+                            env.debSourceBuild_result = "SUCCESS" // This will only run if the steps above succeeded
+                        }
+                    }
+                    post {
+                        always {
+                            bbcGithubNotify(context: "deb/sourceBuild", status: env.debSourceBuild_result)
+                        }
+                    }
+                }
+            }
+        }
+        stage ("Build with pbuilder") {
+            steps {
+                bbcGithubNotify(context: "deb/packageBuild", status: "PENDING")
+                // Build for all supported platforms and extract results into workspace
+                bbcParallelPbuild(stashname: "deb_dist", dists: bbcGetSupportedUbuntuVersions(), arch: "amd64")
+            }
+            post {
+                success {
+                    archiveArtifacts artifacts: "_result/**"
+                }
+                always {
+                    // currentResult is governed by the outcome of the pbuilder steps at this point, so we can use it
+                    bbcGithubNotify(context: "deb/packageBuild", status: currentBuild.currentResult)
```

Comment:
After a discussion with @j616 it might be better to use the variable in a script pattern seen elsewhere, since this is the only place this method shows up. More generally, https://github.com/bbc/rd-apmm-groovy-ci-library/commit/0d8a57c60e06274f4a3681a46d88bd049a1243bd contains some changes to restructure the parallel stages and build wheels and packages together. Up to you whether to pull those in.